packages/transaction-controller/src/TransactionController.ts

```diff
@@ -800,11 +802,15 @@ export class TransactionController extends BaseController<
       this.getNetworkState();
     const { chainId: currentChainId } = providerConfig;
     const newTransactions = this.state.transactions.filter(
-      ({ networkID, chainId }) => {
+      ({ networkID, chainId, transaction }) => {
         // Using fallback to networkID only when there is no chainId present. Should be removed when networkID is completely removed.
         const isCurrentNetwork =
           chainId === currentChainId ||
           (!chainId && networkID === currentNetworkID);
+        if (address) {
+          const isFromAddress = transaction.from === address;
+          return !(isCurrentNetwork && isFromAddress);
```

Comment:
This code is assuming we always care about the network but should we be treating the filter arguments as independent?

Perhaps a simpler and more scalable algorithm would be:

```
const isValidNetwork = ignoreNetwork ||
    chainId === currentChainId || (!chainId && networkID === currentNetworkID);

if(!isValidNetwork) return false;

const isValidAddress = !address || transaction.from?.toLowerCase() === address.toLowerCase();

if(!isValidAddress) return false;

return true;
```

Reply:
Makes sense to early return on that unmatching networks, fixed